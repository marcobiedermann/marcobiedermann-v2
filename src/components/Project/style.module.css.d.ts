/* eslint-disable camelcase */

export const project: string;
export const project__caption: string;
export const project__figure: string;
export const project__link: string;
export const project__title: string;
