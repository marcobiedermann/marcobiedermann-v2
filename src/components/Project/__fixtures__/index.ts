const project = {
  id: '6c5a5baa-8c8b-5580-8061-b1cad23712cc',
  slug: 'demo-project',
  thumbnail: {
    fluid: {
      aspectRatio: 1.333333333,
      sizes: '(max-width: 800px) 100vw, 800px',
      src: 'https://via.placeholder.com/800x600',
      srcSet: 'https://via.placeholder.com/800x600',
    },
  },
  title: 'Demo Project',
};

export default project;
