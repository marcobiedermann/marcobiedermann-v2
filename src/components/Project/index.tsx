import { Link } from 'gatsby';
import Img, { FluidObject } from 'gatsby-image';
import React from 'react';
import styles from './style.module.css';

export interface ProjectProps {
  id: string;
  slug: string;
  thumbnail: {
    fluid: FluidObject;
  };
  title: string;
}

const Project: React.FC<ProjectProps> = (props) => {
  const { slug, thumbnail, title, ...otherProps } = props;

  return (
    <div {...otherProps}>
      <figure className={styles.project__figure}>
        <Link className={styles.project__link} to={`/projects/${slug}`}>
          {thumbnail && <Img fluid={thumbnail.fluid} />}
          <figcaption className={styles.project__caption}>
            <div className={styles.project__title}>
              <h3>{title}</h3>
            </div>
          </figcaption>
        </Link>
      </figure>
    </div>
  );
};

export default Project;
