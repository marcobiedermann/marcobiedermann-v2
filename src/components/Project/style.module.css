.project__caption {
  background-color: rgba(0, 0, 0, 0.9);
  bottom: 0;
  color: #fafafa;
  display: grid;
  left: 0;
  opacity: 0;
  overflow: hidden;
  padding: 1.5em;
  place-items: center;
  position: absolute;
  right: 0;
  text-align: center;
  top: 0;
  transition: opacity 0.5s;
}

.project__figure {
  position: relative;
}

.project__title {
  transform: translateY(20%);
  transition: transform 0.75s;
}

.project__link:focus .project__caption,
.project__link:hover .project__caption {
  opacity: 1;
}

.project__link:focus .project__title,
.project__link:hover .project__title {
  transform: translateY(0);
}
