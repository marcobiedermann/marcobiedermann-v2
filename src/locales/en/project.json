{
  "agency": "Agency",
  "client": "Client",
  "colors": {
    "title": "Colors"
  },
  "homepage": {
    "title": "Homepage"
  },
  "mobile": {
    "description": "With a small display comes great responsibilities",
    "title": "Mobile View"
  },
  "more work": "See more Work",
  "tablet": {
    "title": "Tablet View"
  },
  "role": "Role",
  "responsive": {
    "description": "A website that displays nicely on desktop and on your phone and tablet",
    "title": "Responsive Design"
  },
  "typography": {
    "title": "Typography"
  },
  "website": "Visit Website"
}
