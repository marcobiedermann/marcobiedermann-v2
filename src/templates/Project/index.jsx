import { graphql, Link } from 'gatsby';
import Img from 'gatsby-image';
import PropTypes from 'prop-types';
import React from 'react';
import Column from '../../components/Column';
import Grid from '../../components/Grid';
import Layout from '../../components/Layout';
import Pagination from '../../components/Pagination';
import Row from '../../components/Row';
import Section from '../../components/Section';

const ProjectTemplate = props => {
  const {
    data: {
      allContentfulProject: { edges },
      contentfulProject: {
        agency,
        client,
        colors,
        desktop,
        homepage,
        mobile,
        primaryColor,
        tablet,
        role,
        title,
        typography,
      },
    },
    pageContext: { previous, next },
  } = props;

  const paginationRoutes = [];

  if (previous) {
    paginationRoutes.push({
      name: 'Previous',
      path: `/projects/${previous.slug}`,
    });
  }

  if (next) {
    paginationRoutes.push({
      name: 'Next',
      path: `/projects/${next.slug}`,
    });
  }

  return (
    <Layout>
      {desktop && (
        <Section isFull>
          <Grid>
            <Img fluid={desktop.fluid} />
          </Grid>
        </Section>
      )}

      <Section>
        <Grid>
          <h1>{title}</h1>
          <ul>
            {client && (
              <li>
                <strong>Client:</strong> {client}
              </li>
            )}

            {agency && (
              <li>
                <strong>Agency:</strong> {agency}
              </li>
            )}

            {role && (
              <li>
                <strong>Role:</strong> {role}
              </li>
            )}
          </ul>
        </Grid>
      </Section>

      {colors && (
        <Section>
          <Grid>
            <h2>Colors</h2>
            <Img fluid={colors.fluid} />
          </Grid>
        </Section>
      )}

      {typography && (
        <Section>
          <Grid>
            <h2>Typography</h2>
            <Row>
              {typography.map(font => (
                <Column key={font.title}>
                  <Img fluid={font.fluid} />
                </Column>
              ))}
            </Row>
          </Grid>
        </Section>
      )}

      {homepage && (
        <Section>
          <Grid>
            <h2>Homepage</h2>
            <Img fluid={homepage.fluid} />
          </Grid>
        </Section>
      )}

      <Section
        style={{
          backgroundColor: primaryColor,
        }}
      >
        <Grid>Responsive Design</Grid>
      </Section>

      {tablet && (
        <Section>
          <Grid>
            <h2>Tablet view</h2>
            <Img fluid={tablet.fluid} />
          </Grid>
        </Section>
      )}

      {mobile && (
        <Section>
          <Grid>
            <h2>Mobile View</h2>
            <p>With a small display comes great responsibilities</p>
            <Img fluid={mobile.fluid} />
          </Grid>
        </Section>
      )}

      <Section>
        <Grid>
          <Pagination routes={paginationRoutes} />
        </Grid>
      </Section>

      <Section>
        <Grid>
          <ul>
            {edges.map(edge => (
              <li key={edge.node.id}>
                <Link to={`/projects/${edge.node.slug}`}>
                  <img src={edge.node.logo.file.url} alt={edge.node.logo.title} />
                </Link>
              </li>
            ))}
          </ul>
        </Grid>
      </Section>
    </Layout>
  );
};

ProjectTemplate.propTypes = {
  data: PropTypes.shape({
    allContentfulProject: PropTypes.shape({
      edges: PropTypes.arrayOf(
        PropTypes.shape({
          node: PropTypes.shape({
            logo: PropTypes.shape({
              file: PropTypes.shape({
                url: PropTypes.string,
              }),
              title: PropTypes.string,
            }),
            id: PropTypes.string,
            slug: PropTypes.string,
          }),
        }),
      ),
    }),
    contentfulProject: PropTypes.shape({
      agency: PropTypes.string,
      client: PropTypes.string,
      colors: PropTypes.shape({
        fluid: PropTypes.shape(),
      }),
      desktop: PropTypes.shape({
        fluid: PropTypes.shape(),
      }),
      homepage: PropTypes.shape({
        fluid: PropTypes.shape(),
      }),
      mobile: PropTypes.shape({
        fluid: PropTypes.shape(),
      }),
      primaryColor: PropTypes.string,
      role: PropTypes.string,
      tablet: PropTypes.shape({
        fluid: PropTypes.shape(),
      }),
      title: PropTypes.string,
      typography: PropTypes.arrayOf(
        PropTypes.shape({
          fluid: PropTypes.shape(),
          title: PropTypes.string,
        }),
      ),
    }),
  }),
  pageContext: PropTypes.shape({
    previous: PropTypes.shape({
      slug: PropTypes.string,
    }),
    next: PropTypes.shape({
      slug: PropTypes.string,
    }),
  }),
};

ProjectTemplate.defaultProps = {
  data: null,
  pageContext: null,
};

export const pageQuery = graphql`
  query ProjectBySlug($slug: String!) {
    allContentfulProject(filter: { logo: { id: { ne: null } } }) {
      edges {
        node {
          id
          logo {
            file {
              url
            }
            title
          }
          slug
        }
      }
    }
    contentfulProject(slug: { eq: $slug }) {
      agency
      client
      colors {
        fluid {
          ...GatsbyContentfulFluid_withWebp_noBase64
        }
      }
      desktop {
        fluid {
          ...GatsbyContentfulFluid_withWebp_noBase64
        }
      }
      homepage {
        fluid {
          ...GatsbyContentfulFluid_withWebp_noBase64
        }
      }
      id
      mobile {
        fluid {
          ...GatsbyContentfulFluid_withWebp_noBase64
        }
      }
      primaryColor
      role
      slug
      tablet {
        fluid {
          ...GatsbyContentfulFluid_withWebp_noBase64
        }
      }
      tags
      title
      typography {
        fluid {
          ...GatsbyContentfulFluid_withWebp_noBase64
        }
        title
      }
      url
    }
  }
`;

export default ProjectTemplate;
