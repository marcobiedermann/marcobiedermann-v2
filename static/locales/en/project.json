{
  "agency": "Agency",
  "client": "Client",
  "colors": {
    "title": "Colors"
  },
  "homepage": {
    "title": "Homepage"
  },
  "mobile": {
    "description": "With a small display comes great responsibilities",
    "title": "Mobile View"
  },
  "more work": "See more Work",
  "tablet": {
    "title": "Tablet View"
  },
  "role": "Role",
  "responsive": "Responsive Design",
  "typography": {
    "title": "Typography"
  }
}
