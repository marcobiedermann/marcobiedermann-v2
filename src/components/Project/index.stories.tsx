/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Project from '.';
import project from './__fixtures__';

storiesOf('Project', module).add('default', () => <Project {...project} />);
