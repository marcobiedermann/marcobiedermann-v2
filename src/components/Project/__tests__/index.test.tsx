/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Project from '..';
import project from '../__fixtures__';

describe('Project component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Project {...project} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
