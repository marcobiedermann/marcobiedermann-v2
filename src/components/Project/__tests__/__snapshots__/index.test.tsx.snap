// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component should render correctly 1`] = `
<DocumentFragment>
  <div
    id="6c5a5baa-8c8b-5580-8061-b1cad23712cc"
  >
    <figure>
      <a
        href="/projects/demo-project"
      >
        <figcaption>
          <div>
            <h3>
              Demo Project
            </h3>
          </div>
        </figcaption>
      </a>
    </figure>
  </div>
</DocumentFragment>
`;
